src/kw_params.jl

```diff
@@ -0,0 +1,36 @@
+using Match, NamedTuples
+"""
+Macro which takes assignment arguments (e.g., "a=2") and gives us a Named Tuple constructor that
+(a) has those assignments as defaults, and (b) takes keyword redefinitions. 
+
+This depends on NamedTuples.jl and Match.jl for v0.6.
+"""
+macro kw_params(args...)
```

Comment:
I used the same macro `@with_kw` in my PR.  Any reason you decided to add a second one?

Reply:
I think, when I did this, the issue was I didn't want to overload `@with_kw(typedef)`, the one-arg method you had. And since the syntax was up in the air (is it going to be `@kw_params (blahblahblah)` or `@kw_params(blahblahblah)`, figured this would be best.